src/g_game.c

```diff
@@ -219,20 +219,16 @@ int    bodyqueslot, bodyquesize, default_bodyquesize; // killough 2/8/98, 10/98
 
 static int next_weapon = 0;
 
-static const struct
-{
-    weapontype_t weapon;
-    weapontype_t weapon_num;
-} weapon_order_table[] = {
-    { wp_fist,            wp_fist },
-    { wp_chainsaw,        wp_fist },
-    { wp_pistol,          wp_pistol },
-    { wp_shotgun,         wp_shotgun },
-    { wp_supershotgun,    wp_shotgun },
-    { wp_chaingun,        wp_chaingun },
-    { wp_missile,         wp_missile },
-    { wp_plasma,          wp_plasma },
-    { wp_bfg,             wp_bfg }
+static const weapontype_t weapon_order_table[] = {
```

Comment:
Now this always allows to switch from chainsaw back to fists, right? I think this was the reason for the two-fold table.

Reply:
No, if the player has a chainsaw, it will only switch to firsts with berserk. See
https://github.com/fabiangreffrath/woof/blob/ff6c43e238087fdc455efde074c877a19ab2fa6d/src/g_game.c#L737

Reply:
Ah, right, this check comes later, so we don't need it in the prev/next table. Thanks!

Reply:
Still not sure if it works properly. @MrAlaux could you please check?

Reply:
> Still not sure if it works properly. @MrAlaux could you please check?

`doom_weapon_toggles == 0` works properly now: both buttons can switch to all weapons, except the non-Berserk Fist

As for `doom_weapon_toggles == 1`, or vanilla complevel (in which case `doom_weapon_toggles` should take no effect):
\- _Previous_ can switch to all weapons, except the non-Berserk Fist
\- _Next_ omits the Fist (regardless of Berserk) and Shotgun

Reply:
I have disabled `doom_weapon_toggles` for the next weapon here: https://github.com/fabiangreffrath/woof/pull/1785/commits/3ac46d970360b7784c6d3d86895021e99df47b7f What do you think?

Reply:
Sure enough, the prev/next-weapon buttons seem to completely ignore `doom_weapon_toggles` now, and both can switch to all weapons (except, as you may guess, the non-Berserk Fist). I think this works best.

FWIW, vanilla complevel remains unaffected.

It should be obvious, but I should have clarified that the Fist and Shotgun are only skipped when you have the Chainsaw and SSG respectively.